File: standalone/dzen2-hlwm/lua/pipehandler.01-init.lua
-- luaposix available in AUR
local posix = require "posix"

local common = require('.common')
local helper = require('.helper')
local output = require('.output')

local _M = {}

-- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
-- pipe

function _M.content_init(monitor, pipe_dzen2_out)
    -- initialize statusbar before loop
    output.set_tag_value(monitor)
    output.set_windowtitle('')

    local text = output.get_statusbar_text(monitor)
    pipe_dzen2_out:write(text .. "\n")
    pipe_dzen2_out:flush()
end

function _M.run_dzen2(monitor, parameters) 
    local command_out    = 'dzen2 ' .. parameters .. ' -p'
    local pipe_dzen2_out = assert(io.popen(command_out, 'w'))
    
    _M.content_init(monitor, pipe_dzen2_out)
    pipe_dzen2_out:close()
end

function _M.detach_dzen2(monitor, parameters)
    local pid = posix.fork()

    if pid == 0 then -- this is the child process
        _M.run_dzen2(monitor, parameters)
    else             -- this is the parent process
        -- nothing
    end
end

function _M.detach_transset()
    local pid = posix.fork()

    if pid == 0 then -- this is the child process
        common.sleep(1)
        os.execute('transset .8 -n dzentop >/dev/null') 
    else             -- this is the parent process
        -- nothing
    end
end


-- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
-- return

return _M

File: standalone/dzen2-hlwm/php/pipehandler.01-init.php
<?php # using PHP7

require_once(__DIR__.'/output.php');

# ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----
# pipe

function content_init($monitor, $pipe_dzen2_out)
{   
    set_tag_value($monitor);
    set_windowtitle('');
        
    $text = get_statusbar_text($monitor);
    fwrite($pipe_dzen2_out, $text."\n");
    flush();
}

function run_dzen2($monitor, $parameters) 
{ 
    $command_out    = "dzen2 $parameters -p";
    $pipe_dzen2_out = popen($command_out, 'w');

    content_init($monitor, $pipe_dzen2_out);
    pclose($pipe_dzen2_out);
}

function detach_dzen2($monitor, $parameters)
{ 
    $pid_dzen2 = pcntl_fork();
    
    switch($pid_dzen2) {         
    case -1 : // fork errror         
        die('could not fork');
    case 0  : // we are the child
        run_dzen2($monitor, $parameters); 
        break;
    default : // we are the parent             
        return $pid_dzen2;
    }    
}

function detach_transset() 
{ 
    $pid_transset = pcntl_fork();
    if ($pid_transset == 0) { 
        sleep(1);
        system('transset .8 -n dzentop >/dev/null');
    }
}
